pkg/knuu/instance_old.go

```diff
@@ -199,48 +199,8 @@ func (i *Instance) SetStartupProbe(startupProbe *v1.Probe) error {
 }
 
 // Deprecated: Use the new package knuu instead.
-func (i *Instance) AddSidecar(sidecar *Instance) error {
-	return i.Instance.AddSidecar(&sidecar.Instance)
-}
-
-// Deprecated: Use the new package knuu instead.
-func (i *Instance) SetOtelCollectorVersion(version string) error {
-	return i.Instance.SetOtelCollectorVersion(version)
-}
-
-// Deprecated: Use the new package knuu instead.
-func (i *Instance) SetOtelEndpoint(port int) error {
-	return i.Instance.SetOtelEndpoint(port)
-}
-
-// Deprecated: Use the new package knuu instead.
-func (i *Instance) SetPrometheusEndpoint(port int, jobName, scapeInterval string) error {
-	return i.Instance.SetPrometheusEndpoint(port, jobName, scapeInterval)
-}
-
-// Deprecated: Use the new package knuu instead.
-func (i *Instance) SetJaegerEndpoint(grpcPort, thriftCompactPort, thriftHttpPort int) error {
-	return i.Instance.SetJaegerEndpoint(grpcPort, thriftCompactPort, thriftHttpPort)
-}
-
-// Deprecated: Use the new package knuu instead.
-func (i *Instance) SetOtlpExporter(endpoint, username, password string) error {
-	return i.Instance.SetOtlpExporter(endpoint, username, password)
-}
-
-// Deprecated: Use the new package knuu instead.
-func (i *Instance) SetJaegerExporter(endpoint string) error {
-	return i.Instance.SetJaegerExporter(endpoint)
-}
-
-// Deprecated: Use the new package knuu instead.
-func (i *Instance) SetPrometheusExporter(endpoint string) error {
-	return i.Instance.SetPrometheusExporter(endpoint)
-}
-
-// Deprecated: Use the new package knuu instead.
-func (i *Instance) SetPrometheusRemoteWriteExporter(endpoint string) error {
-	return i.Instance.SetPrometheusRemoteWriteExporter(endpoint)
+func (i *Instance) AddSidecar(ctx context.Context, sc instance.SidecarManager) error {
```

Comment:
It would be great to override the default image used in the sidecar if you want to test another version.

Reply:
When user creates the sidecar object, they can assign any image they want, if not the default image will be considered. Is that ok?

Reply:
Just double checked it, for some sidecars we do have it. Let me modify the others code so we can have customized image if user wants.

Reply:
a fix added